client/my-sites/checkout/index.js

```diff
@@ -299,7 +300,7 @@ export default function () {
 	page(
 		`/checkout/:domainOrProduct`,
 		setLocaleMiddleware(),
-		redirectLoggedOut,
+		redirectLoggedOutIfNotFromMyJetpack,
```

Comment:
I understand this solution 👍 

I was curious aboyt having both `redirectLoggedOut` and `redirectMyJetpack` (or similar) as two different middlewares - with the My Jetpack one being called first. 

The My Jetpack middleware could redirect to the siteless checkout path as needed (if it detected a user was logged out ) and pass the rest of the requests through to the `redirectLoggedOut` middleware 🤔 

WDYT?